docs/fsharp/tutorials/getting-started/getting-started-vscode.md

```diff
@@ -1,4 +1,269 @@
+
+[!code-fsharp[ToPigLatin](../../../../samples/snippets/fsharp/getting-started/to-pig-latin.fsx#L8-L8)]
+
+The flow of `toPigLatin` is thus:
+
+1. Convert the input word into a character array.
+2. Check if the first character is a vowel.
+
+    a. If it is, attach "yay" to the end of the word.
+
+    b. Otherwise, move that first character to the end of the word and add "ay" to it.
+
+There's one final thing to notice about this: there's no explicit instruction to return from the function, unlike many other languages out there.  This is because F# is Expression-based, and the last expression in the body of a function is the return value.  Because `if..then..else` is itself an expression, the body of the `then` block or the body of the `else` block will be returned depending on the input value.
+
+## Making your script code part of your source code
+
+The previous sections in this article demonstrated a common first step in writing F# code: writing an initial function and executing it interactively with FSI.  This is known as REPL-driven development, where REPL stands for "Read-Evaluate-Print Loop".  It's a great way to tweak with functionality until you have something working.
+
+The next step is to incorporate code in your F# script file into F# source code.  To begin, open `ClassLibraryDemo.fs`.  You'll notice that some code is already in there.  Go ahead and delete the class definition, but make sure to leave the [`namespace`](../../language-reference/namespaces.md) declaration at the top.
+
+Next, create a new [`module`](../../language-reference/modules.md) called `PigLatin` and copy the `toPigLatin` function into it as such:
+
+[!code-fsharp[ToPigLatin](../../../../samples/snippets/fsharp/getting-started/pig-latin.fs#L1-L16)]
+
+This is one of the many ways you can organize functions in F# code, and a common approach if you also want to call your code from C# or Visual Basic projects.
+
+Next, open the `Script.fsx` file again, and delete the entire `toPigLatin` function, but make sure to keep the following two lines in the file:
+
+```
+#load "ClassLibraryDemo.fs"
+open ClassLibraryDemo
+```
+
+Those two lines will load the `ClassLibraryDemo.fs` file and `open` the `ClassLibraryDemo` namespace, just like if you were to reference this code in another project.  Note that you can also open FSI with `>FSI: Start`, then type those two lines with `;;` as the delimiter to accomplish the same thing.
+
+Next, call the function with the `PigLatin` module that you defined earlier:
+
+```
+> PigLatin.toPigLatin "banana";;
+val it : string = "ananabay"
+> PigLatin.toPigLatin "apple";;
+val it : string = "appleyay"
+```
+
+Success!  You get the same results as before, but now loaded from an F# source file.  The major difference here is that F# source files are compiled into assemblies which can be executed anywhere, not just in FSI.
+
+## Summary
```

Comment:
It may be worth mentioning somewhere that for OSX/nix, you can get F# debugging in VSCode using mono-debugger extension. Would probably add whole section on it, showing how to setup it (as it's not super straightforwards) and some screens from debuggin.